src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1224,7 +1225,7 @@ private long EnforceConfigurationFor(DocumentsOperationContext context, string i
 
                 var prevRevisionsCount = GetRevisionsCount(context, id);
                 var configuration = GetRevisionsConfiguration(collectionName.Name);
-
+                configuration.MaxRevisionsToDeleteUponDocumentUpdate = 0;
```

Comment:
Actually, when we enforce the configuration and we have a lot to delete, we might hold the write lock for a long time.
So let's remove that line, but need to make sure that enforcing will eventually clean all unwanted revisions.